server/src/main/java/io/druid/server/metrics/MetricsModule.java

```diff
@@ -107,39 +106,37 @@ public MonitorScheduler getMonitorScheduler(
 
   @Provides
   @ManageLifecycle
-  public JvmMonitor getJvmMonitor(Properties props)
+  public JvmMonitor getJvmMonitor(
+      DataSourceTaskIdHolder dataSourceTaskIdHolder
```

Comment:
what happens for nodes that don't bind DataSourceTaskIdHolder? is this throwing NPEs?